plaidml/bridge/openvino/src/plaidml_plugin/plaidml_op_permute.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vector>
+
+#include "plaidml/op/op.h"
+
+#include "plaidml_op_permute.hpp"
+#include "plaidml_util.hpp"
+
+void OpPermute::run(const plaidml::edsl::Tensor& I, plaidml::edsl::Tensor& O) {
+  const auto& order = layer_->GetParamAsInts("order");
+  auto layout = layer_->insData.front().lock()->getTensorDesc().getLayout();
+  if (layout == NCHW) {
+    // NB: Input data in the plugin is always in the NHWC format
+    // If the IE has got NCHW, the axes are also for NCHW,
+    // so we need to change them for NHWC case
+    // after the transpositon input has got NCHW format
+    // and we again put conversion from NCHW to NHWC
+    std::vector<int64_t> to_nhwc = {0, 3, 1, 2};
```

Comment:
We'll need a fancier solution to channel ordering if OpenVINO ever uses more or fewer than 4 dims.